File: unix/boot/spp/rpp/rppfor/ifparm.f
      integer function ifparm (strng)
      integer strng (100)
      integer c
      external index
      integer i, index, type
      c = strng (1)
      if (.not.(c .eq. -12 .or. c .eq. -13 .or. c .eq. -11 .or. c .eq. -
     *14 .or. c .eq. -10))goto 23000
      ifparm = 1
      goto 23001
23000 continue
      ifparm = 0
      i = 1
23002 if (.not.(index (strng (i), 36) .gt. 0))goto 23004
      i = i + index (strng (i), 36)
      if (.not.(type (strng (i)) .eq. 48))goto 23005
      if (.not.(type (strng (i + 1)) .ne. 48))goto 23007
      ifparm = 1
      goto 23004
23007 continue
23005 continue
23003 goto 23002
23004 continue
23001 continue
      return
      end

File: unix/boot/spp/rpp/rpprat/ifparm.r
#-h-  ifparm			  689  local   12/01/80  15:54:11
# ifparm - determines if the defined symbol has arguments in its
   include  defs
# definition.  This effects how the macro is expanded.

   integer function ifparm (strng)
   character strng (ARB)

   character c

   external index
   integer i, index, type

   c = strng (1)
   if (c == INCTYPE | c == SUBTYPE | c == IFTYPE | c == ARITHTYPE |
    c == MACTYPE)
      ifparm = YES
   else {
      ifparm = NO
      for (i = 1; index (strng (i), ARGFLAG) > 0; ) {
	 i = i + index (strng (i), ARGFLAG) # i points at char after ARGFLAG
	 if ((type (strng (i)) == DIGIT)
	   & (type (strng (i + 1)) != DIGIT)) {
	    ifparm = YES
	    break
	 }
      }
   }
   return
   end
